### ISSUE TITLE
Getting Video Create Date from iPhone Videos, when it is the 6th atom.

### ISSUE THREAD

**Issue by leenooks** [JamesHeinrich/getID3#139]
Hi, I've been playing with your tool to help me sort the 1000's of videos that I have from my wifes and my iPhone. One thing I'm trying to get is the original video creation time and I can see the data in the videos.

Doing a dump of my GetID3 object on a video file, I can see the "moov meta" dump 6 keys (make, model, software, etc) and key 6 is "com.apple.quicktime.creationdate". However, "moov meta ilst" is only showing 5 entries with data correctly and entry number 6 looks like this:

```
 [5] => Array
  (
    [hierarchy] => moov meta ilst 
    [name] => 
    [size] => 48
    [offset] => 4509390
    [data] => (data2017-07-22T16:06:06+1000
  )
```

This entry looks different to the others in that it doesnt have any "subatoms" it has a "data" key - it's name is also blank. The corresponding key field looks like this:

```
[6] => Array
       (
         [key_size] => 40
         [key_namespace] => mdta
         [key_value] => com.apple.quicktime.creationdate
       )
```

(Note that "moov meta keys" starts at index 1, whereas "moov meta ilist" starts at index 0).

Is there a reason that this value is garbled? Anyway to get it into the info -> tags keys?

**Comment by JamesHeinrich** [comment]
If you could provide a small sample file I can take a closer look

**Comment by leenooks** [comment]
OK, here is another file - this one works out the creationdate OK (it appears to be the 5th field in this video) - however still, the 6th field (in this video "signature") is garbled - so I guess it is not an issue working out the date, but rather handling the fields and the "last one".
I can share the original video as I referenced above as well if needed.

[sample.zip](https://github.com/JamesHeinrich/getID3/files/1602325/sample.zip)

**Comment by leenooks** [comment]
For reference, this is what getID renders for me:

* keys
```
                                                            [entry_count] => 6
                                                            [keys] => Array
                                                                (
                                                                    [1] => Array
                                                                        (
                                                                            [key_size] => 44
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.quicktime.location.ISO6709
                                                                        )

                                                                    [2] => Array
                                                                        (
                                                                            [key_size] => 32
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.quicktime.make
                                                                        )

                                                                    [3] => Array
                                                                        (
                                                                            [key_size] => 33
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.quicktime.model
                                                                        )

                                                                    [4] => Array
                                                                        (
                                                                            [key_size] => 36
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.quicktime.software
                                                                        )

                                                                    [5] => Array
                                                                        (
                                                                            [key_size] => 40
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.quicktime.creationdate
                                                                        )

                                                                    [6] => Array
                                                                        (
                                                                            [key_size] => 46
                                                                            [key_namespace] => mdta
                                                                            [key_value] => com.apple.photos.originating.signature
                                                                        )
```

and the last 2 values

```

                                                                    [4] => Array
                                                                        (
                                                                            [hierarchy] => moov meta ilst ^@^@^@^E
                                                                            [name] => 5
                                                                            [size] => 48
                                                                            [offset] => 5744
                                                                            [subatoms] => Array
                                                                                (
                                                                                    [0] => Array
                                                                                        (
                                                                                            [hierarchy] => moov meta ilst ^@^@^@^E data
                                                                                            [name] => data
                                                                                            [size] => 40
                                                                                            [offset] => 5752
                                                                                            [language] => ^@^@
                                                                                            [unknown] => 0
                                                                                            [data] => 2018-01-04T12:55:44+1100
                                                                                            [key_name] => com.apple.quicktime.creationdate
                                                                                        )

                                                                                )

                                                                        )

                                                                    [5] => Array
                                                                        (
                                                                            [hierarchy] => moov meta ilst ^@^@^@^F
                                                                            [name] => ^@^@^@^F
                                                                            [size] => 52
                                                                            [offset] => 5792
                                                                            [data] => ^@^@^@,data^@^@^@^A^@^@^@^@AbZ8pDGtHddmCfoiiLBHrDqCdtBo
                                                                        )

```

**Comment by JamesHeinrich** [comment]
These fields are already parsed into [quicktime][comments] under the same name but with "com.apple.quicktime." removed, so for example your creation date is found at [quicktime][comments][creationdate][0]
![quicktime](https://user-images.githubusercontent.com/3483435/34581660-4c5d0256-f15f-11e7-997a-2c835cf6b34b.png)

**Comment by leenooks** [comment]
No, I think you've missed it.

This is not a problem about getting creationdate per-se. Its about getting the value in the 6th field (or probably "last" field).

In the sample I gave you, the value was garbled for "signature". In my other images, the last field is the creationdate, and thus I cannot retrieve it.

Here is another sample:
[sample.zip](https://github.com/JamesHeinrich/getID3/files/1604647/sample.zip)

**Comment by JamesHeinrich** [comment]
I see what you mean now.
After some puzzling, I found that it wasn't necessarily the last item, but any items beyond the 5th item -- handling was hardcoded for items up to 5, but not beyond. In your examples there just happened to be 6 entries so it looked like the "last" item.  The code as it stands now should handle up to 255 entries, if someone finds an example with more than that let me know :)

Fixed in https://github.com/JamesHeinrich/getID3/commit/d5a430e3e1b71a1a3730b05be7f8a650ab36dca1 (the commit is kind of ugly due to whitespace changes from moving a 1000+ lines inside an if statement)
